File: Python/q3.py

def primeFactor(n):
    factor = 2
    while n > 1:
        if n % factor == 0:
            n //= factor
        else:
            factor += 1
    return factor

number = int(input("Enter the number: "))

result = primeFactor(number)
print(result)

File: Dart/q3.dart
import 'dart:io';

int primeFactor(int number) {
  int factor = 2;
  while (number > 1) {
    if (number % factor == 0) {
      number ~/= factor;
    } else {
      factor++;
    }
  }
  return factor;
}

void main() {
  print("Enter the number to find largest prime factor");
  int number = int.parse(stdin.readLineSync()!);
  int result = primeFactor(number);
  print(result);
}
